python/tank_vendor/shotgun_authentication/user.py

```diff
@@ -0,0 +1,392 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+"""
+Defines the supported types of authentication methods with Shotgun. You can
+either authenticate with a session token with the SessionUser class or with an
+api key with the ScriptUser class. This module is meant to be used internally.
+"""
+
+import pickle
+from .shotgun_wrapper import ShotgunWrapper
+
+from . import session_cache
+from .errors import CachingVolatileUserException
+
+
+_shotgun_instance_factory = ShotgunWrapper
```

Comment:
A factory is an object or method that allows to create instances of an object. The class object fits that description, plus in some iterations of the code some unit tests were actually assigning a function to that variable, so it would definitely fit that description.